### ISSUE TITLE
Feature Requests - General Inquiry

### ISSUE THREAD

**Issue by mikebronner** [atom/ide-php#38]
I am noticing that the following functionalities are not working at this time (be it through the language server or this package):

- namespace auto-complete hints while typing a a use statement are missing [issue #18]
- classes and methods are not being provided to auto-complete, and only sometimes if they are in the same file. It seems to only auto-complete previously typed items consistently [mentioned in issue #30]
- use statements are not being added when autocompleting a class (in the rare event that it actually provides a class hint in the auto-complete list) [issue #36]

Are these issues we need to raise with the FelixFBecker language server, or should I create individual issues to track these here? If these are related to ide-php, I would love to dig in and help out with some PRs, I just wanted to see where these should go first.

The main inquiry here is to see if I can help out with this functionality through PRs, or if this is outside of this package's control. And if so, could we switch to a language server that supports these features?

Thanks :)

**Comment by damieng** [comment]
If the features aren't supported in the FelixFBecker langauge server you'll need to file issues there.

If they are supported but are somehow just not working in Atom file them here - we'll need to investigate/wire-up as appropriate.

**Comment by mikebronner** [comment]
Can you elaborate on the above three examples which would be where?

**Comment by damieng** [comment]
Without investigating all of those and looking at the debug messages in the console I couldn't be sure.  

It might be worth trying the FelixFBecker language server extension in VSCode (It's called php-intellisense).  If it works there it's probably a problem in our code if it doesn't then it's a current limitation of the language server.

**Comment by mikebronner** [comment]
Yea, that language server caused lots of issues in vscode and was unusable for me. I reverted to php intelliphense (in vscode), which did all of the above items. Then I saw this project, and reverted back to atom, as it seems much more streamlined.

Is changing language servers a possibility?

**Comment by damieng** [comment]
If there was a better language server we would consider it.

Alternatively somebody could create an alternative ide package - ide-php-bobslanguageserver - that powered that. It's quite easy to do - the readme for atom-languageclient has the details or you can look at the source for this project.

**Comment by 3Descape** [comment]
Seems to me like it's more package related after testing it in VSCode. I can recommend that sever:https://gitlab.com/php-integrator/core . This server seems to be the backend for this package https://github.com/felixfbecker/php-language-server and it does a really solid job. It's what I currently use, untill this package gets stable enough.

**Comment by damieng** [comment]
We're using the felixfbecker language server here in ide-php

**Comment by 3Descape** [comment]
I'm so sorry.. I messed up all the opened tabs on my browser. This is the package: https://github.com/php-integrator/atom-base and this seems to be the server https://gitlab.com/php-integrator/core based on this page: https://php-integrator.github.io/#what-is-it-really

**Comment by mikebronner** [comment]
@3Descape I tried php-integrator as well, and it was not good enough for me to be productive. The auto-completion and other things still felt sub-par. I did have a fairly stable setup that pretty much covered what ide-php is aiming to do, but the downside was that packages kept getting out of whack when atom updated core functionality. And then it would take several weeks before things were working stably again.

@damieng Thanks, I will read up on that. :)
